### PR TITLE
sampling: maintain a single Badger database for process lifetime

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Move Badger outside of the server reload lifecycle, so multiple concurrently existing tail-sampling processors (i.e. overlapping on reload) can access the now process-global, shared, Badger database. The Badger database is opened the first time a tail-sampling processor is created, and is closed when the process exits.

Remove tail-sampling storage directory config; ti is now always "tail_sampling" under the data directory. There is no good reason for it to be configurable, and making it configurable means we can't guarantee a single Badger database for the lifetime of the process.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)
~- [ ] Documentation has been updated~

## How to test these changes

Cannot yet be manually tested. Needs changes to the integration package to test tail-based sampling via Fleet.

## Related issues

None.

### CODE REVIEW

File: x-pack/apm-server/main.go [hunk 1]
```diff
@@ -191,12 +222,30 @@ func wrapRunServer(runServer beater.RunServerFunc) beater.RunServerFunc {
 	}
 }
 
+func closeBadger() error {
+	if badgerDB != nil {
+		return badgerDB.Close()
```
Comment (stuartnelson3) [inline x-pack/apm-server/main.go:227]: does this access need to be coordinated to?

**Review by stuartnelson3 — APPROVED** [review]
one question, but that's all :+1:

File: x-pack/apm-server/main.go (hunk 1 above)
Comment (axw) [reply x-pack/apm-server/main.go:227]: Nope. `closeBadger` is only called at process exit time and never concurrently with `getBadgerDB`. I'll add a comment.

**Comment by stuartnelson3** [comment]
Should we add `test-plan-skip` to this one?

**Comment by axw** [comment]
Yes. Technically it is possible to test via the integration now, but it's not straightforward or supported yet.

**Pull request merged**